### ISSUE TITLE
Why the absolute value of scores?

### ISSUE THREAD

**Issue by cyugao** [allenai/hidden-networks#11]
When I ran the code, I found the gradients weird. I finally realized it was the `scores.abs()` step that flipped the sign of gradients.
Could you explain why we take the absolute value of this? I did notice in the paper there's a footnote.
> To ensure that the scores are positive we take the absolute value.

I think it could also work if we remove the `abs` operation, which matches the actual algorithm described in the paper; maybe that's just a heuristic.

**Comment by mitchellnw** [comment]
I agree it should also work without `abs`, but I don't think this hurts.
